### ISSUE TITLE
CONSTRAINTS not export from sqlite to duckdb

### ISSUE THREAD

**Issue by Victordeleusse** [dridk/sqlite2duckdb#1]
Hello there.
After having used sqlite2duckdb, i have tried to get my constraints from the database through the command 
```
query_pkey = """

                SELECT kcu.table_name, kcu.column_name

                FROM information_schema.table_constraints AS tc

                JOIN information_schema.key_column_usage AS kcu

                ON tc.constraint_name = kcu.constraint_name

                WHERE tc.constraint_type = 'PRIMARY KEY'"""

primary_keys = con.execute(query_pkey).fetchall()
```
But unfortunatly the constraints from the sqlite database haven't been transferred to the duckdb database.

**Comment by dridk** [comment]
Yes, sorry.. I didn't yet implement constraint. I am on it.

**Comment by Victordeleusse** [comment]
@dridk if you need any help on this it would be an honnor to contribute ;) !

**Comment by dridk** [comment]
Of course, you can create au pull requests if you want. It will be helpful

**Comment by dridk** [comment]
Fixed in 0.4.0, just released.

Primary keys and NOT NULL constraints are now carried over, along with indexes. Your exact query returns 12 rows on a converted chinook, where it used to return none:

```
[('albums', 'AlbumId'), ('artists', 'ArtistId'), ('customers', 'CustomerId'), ('employees', 'EmployeeId'), ...]
```

The conversion now recreates each table from the DDL duckdb derives for the attached sqlite database, rather than a `CREATE TABLE AS SELECT` that dropped every constraint on the way.

Still not copied: UNIQUE, FOREIGN KEY and CHECK. duckdb's sqlite extension does not expose those on an attached database, so they would have to be read back from `sqlite_master` and replayed. That is written down in the README as a known limitation.

Closing since the primary key case you reported works now — please open a new issue if you need the other three, or comment here and I will reopen.
